Conflict options in the newlib configuration should not be configured simultaneously

When compiling the following code 
```c
#include <iostream>

int main() {
    std::cout << "hello world" << std::endl;
    return 0;
}

```

with the GCC that compiled from the following config, 
```
CT_CONFIG_VERSION="4"
CT_ARCH_ARM=y
CT_LIBC_NEWLIB_IO_POS_ARGS=y
CT_LIBC_NEWLIB_NANO_FORMATTED_IO=y
CT_CC_LANG_CXX=y
```

it would throw errors:
```
/root/x-tools/arm-unknown-eabi/lib/gcc/arm-unknown-eabi/11.2.0/../../../../arm-unknown-eabi/bin/ld: /root/x-tools/arm-unknown-eabi/lib/gcc/arm-unknown-eabi/11.2.0/../../../../arm-unknown-eabi/lib/libc.a(lib_a-svfwprintf.o): in function `get_arg':
/root/work/.build/arm-unknown-eabi/src/newlib/newlib/libc/stdio/vfwprintf.c:1928: undefined reference to `__chclass'
/root/x-tools/arm-unknown-eabi/lib/gcc/arm-unknown-eabi/11.2.0/../../../../arm-unknown-eabi/bin/ld: /root/work/.build/arm-unknown-eabi/src/newlib/newlib/libc/stdio/vfwprintf.c:1928: undefined reference to `__state_table'
/root/x-tools/arm-unknown-eabi/lib/gcc/arm-unknown-eabi/11.2.0/../../../../arm-unknown-eabi/bin/ld: /root/work/.build/arm-unknown-eabi/src/newlib/newlib/libc/stdio/vfwprintf.c:1928: undefined reference to `__action_table'
collect2: error: ld returned 1 exit status
```
---

After checking the newlib's source code, I found that the undefined symbols are [here](https://github.com/cygwin/cygwin/blob/415fdd4279b85eeec9d54775ce13c5c412451e08/newlib/libc/stdio/vfwprintf.c#L1780) and should be defined [here](https://github.com/cygwin/cygwin/blob/415fdd4279b85eeec9d54775ce13c5c412451e08/newlib/libc/stdio/vfprintf.c#L1966). It seems two macros, `INTEGER_ONLY` and `STRING_ONLY`, should be defined to compile those variables above, and I found these two macros are defined in `newlib/libc/stdio/Makefile.am` [here](https://github.com/cygwin/cygwin/blob/415fdd4279b85eeec9d54775ce13c5c412451e08/newlib/libc/stdio/Makefile.am#L279) and [here](https://github.com/cygwin/cygwin/blob/415fdd4279b85eeec9d54775ce13c5c412451e08/newlib/libc/stdio/Makefile.am#L288) and are dependent on the `vfprintf.c`, but there is a conditional symbol `NEWLIB_NANO_FORMATTED_IO` that would prevent the `vfprintf.c` from compiling [here](https://github.com/cygwin/cygwin/blob/415fdd4279b85eeec9d54775ce13c5c412451e08/newlib/libc/stdio/Makefile.am#L7). 

To sums up, configuring the newlib with `--enable-newlib-nano-formatted-io` and `--enable-newlib-io-pos-args` would cause built GCC cannot compile some IO functions like `std::cout`.

--- a/config/libc/newlib.in
+++ b/config/libc/newlib.in
@@ -55,6 +55,7 @@ config LIBC_NEWLIB_IO_LDBL
 config LIBC_NEWLIB_IO_POS_ARGS
     bool
     prompt "Enable printf-family positional arg support"
+    depends on !LIBC_NEWLIB_NANO_FORMATTED_IO
     help
         Enable printf-family positional arg support.
 
@@ -198,6 +199,29 @@ config LIBC_NEWLIB_NANO_FORMATTED_IO
       functions, designed to lower the size of application on small systems
       with size constraint issues.  This option does not affect wide-char
       formatted I/O functions.
+
+      Some notes about this feature:
+      1) Floating-point support is split out of the formatted I/O code into
+         weak functions which are not linked by default.  Programs that need
+         floating-point I/O support must explicitly request linking of one or
+         both of the floating-point functions: _printf_float or _scanf_float.
+         This can be done at link time using the -u option which can be passed
+         to either gcc or ld.
+
+      2) The following configuration options are ignored for non-wide-char
+         formatted I/O functions, and can be thought of as disabled.
+
+         "Enable printf-family positional arg support"
+         "Enable IOs on C99 formats"
+         "Enable IOs on long long"
+         "Enable IOs on long doubles"
+
+         Additionally, "Enable IOs on floats and doubles" is supported in this
+         specific implementation, one can disable it to further reduce code size.
+         In this case, the floating-point specifiers will not be recognized or
+         handled, and the -u option will not work either.
+
+      Learn more: https://sourceware.org/newlib/README
 
 config LIBC_NEWLIB_EXTRA_CONFIG_ARRAY
     string

